inc/namespace.php

```diff
@@ -80,7 +80,22 @@ function merge_config_settings( array $config, array $overrides ) : array {
 			// Merge module settings together.
 			case 'modules':
 				foreach ( $value as $module => $settings ) {
-					$config[ $key ][ $module ] = array_merge( $config[ $key ][ $module ] ?? [], $settings );
+					$config_settings = $config[ $key ][ $module ] ?? [];
+
+					// Check config settings syntax is valid.
+					if ( ! is_array( $config_settings ) && ! is_bool( $config_settings ) ) {
```

Comment:
🤦‍♂ I should've spotted this - the clue is in the error message, it's actually the 2nd argument in the `array_merge` that's the problem!

So change all of these to be checks against the `$settings` variable.

Reply:
Oops, I didn't notice that too! 🤦‍♂